packages/contracts/src/interest/MultiTokenVault.sol

```diff
@@ -21,37 +21,24 @@ abstract contract MultiTokenVault is IMultiTokenVault, ERC1155, ReentrancyGuard,
     using SafeERC20 for IERC20;
 
     /// @notice Tracks the number of time periods that have elapsed.
-    uint256 public currentTimePeriodsElapsed = 0;
+    uint256 private _currentTimePeriodsElapsed;
 
     /// @notice The ERC20 token used as the underlying asset in the vault.
-    address public immutable asset;
-
-    /// @notice The address of the treasury where deposited assets are transferred.
-    address treasury;
+    IERC20 private immutable _asset;
 
     /// @notice Tracks the total amount of assets deposited in the vault.
     uint256 internal totalDepositedAssets;
 
-    /// @notice The ratio of assets to shares (e.g., 1:1 ratio).
-    uint256 internal immutable ASSET_TO_SHARES_RATIO;
-
-    error MultiTokenVault__UnsupportedFunction(string functionName);
-    error MultiTokenVault__ExceededMaxRedeem(address owner, uint256 depositPeriod, uint256 shares, uint256 max);
-    error MultiTokenVault__RedeemTimePeriodNotSupported(address owner, uint256 period, uint256 redeemPeriod);
+    error MultiTokenVault__ExceededMaxRedeem(address owner, uint256 depositPeriod, uint256 shares, uint256 maxShares);
+    error MultiTokenVault__CallerMissingApprovalForAll();
```

Comment:
Don't you think we should pass some parameters to this error?
And why to remove the `MultiTokenVault__RedeemTimePeriodNotSupported`?